types/mempool/simple.go

```diff
@@ -0,0 +1,93 @@
+package mempool
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	huandu "github.com/huandu/skiplist"
+)
+
+type simpleMempool struct {
```

Comment:
This implementation should be called `nonceOnlyMempool` or similar since nonce data driving ordering.   Let's also add some docs about the prioritization, for instance txs with the smallest nonce are prioritized first.